api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1079,16 +1182,16 @@ type ManagedEtcdStorageSpec struct {
 // PersistentVolumeEtcdStorageSpec is the configuration for PersistentVolume
 // etcd storage.
 type PersistentVolumeEtcdStorageSpec struct {
-	// StorageClassName is the StorageClass of the data volume for each etcd member.
-	//
+	// storageClassName is the StorageClass of the data volume for each etcd member.
 	// See https://kubernetes.io/docs/concepts/storage/persistent-volumes#class-1.
-	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="storageClassName is immutable"
 	// +optional
 	// +immutable
 	StorageClassName *string `json:"storageClassName,omitempty"`
```

Comment:
Why does this need to be a pointer? Is the empty string a valid opinion?

Reply:
Not really, I don't think so. I'll added a TODO to revisit "breaking" changes that consolidate pointer vs non-pointer